workspaces/ballerina/ballerina-visualizer/src/views/EvaluationHistory/SparklineChart.tsx

```
@@ -0,0 +1,297 @@
+    const points = runs
+        .map((_, i) => `${scaleX(i).toFixed(1)},${scaleY(runs[i].passRate).toFixed(1)}`)
+        .join(" ");
+
+    const areaPoints =
+        points +
+        ` ${scaleX(runs.length - 1).toFixed(1)},${(CHART_HEIGHT - PAD_Y).toFixed(1)}` +
+        ` ${PAD_X},${(CHART_HEIGHT - PAD_Y).toFixed(1)}`;
+
+    const gradId = `sparkGrad-${Math.random().toString(36).slice(2)}`;
+
+    const handleMouseEnter = useCallback(
```

Biome (2.4.6)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

(lint/correctness/useHookAtTopLevel)